Sources/Subscription/Managers/SubscriptionManager.swift

```diff
@@ -104,70 +196,229 @@ public final class DefaultSubscriptionManager: SubscriptionManager {
         }
     }
 
-    // MARK: - Environment, ex SubscriptionPurchaseEnvironment
+    // MARK: - Environment
 
     @available(macOS 12.0, iOS 15.0, *) private func setupForAppStore() {
         Task {
             await storePurchaseManager().updateAvailableProducts()
         }
     }
 
-    // MARK: -
+    // MARK: - Subscription
 
-    public func loadInitialData() {
-        Task {
-            if let token = accountManager.accessToken {
-                _ = await subscriptionEndpointService.getSubscription(accessToken: token, cachePolicy: .reloadIgnoringLocalCacheData)
-                _ = await accountManager.fetchEntitlements(cachePolicy: .reloadIgnoringLocalCacheData)
-            }
-        }
-    }
-
-    public func refreshCachedSubscriptionAndEntitlements(completion: @escaping (_ isSubscriptionActive: Bool) -> Void) {
-        Task {
-            guard let token = accountManager.accessToken else { return }
+    public func loadInitialData() async {
 
-            var isSubscriptionActive = false
+        // Attempting V1 token migration
+        // IMPORTANT: This MUST be the first operation executed by Subscription
+        do {
+            if (try await oAuthClient.migrateV1Token()) != nil {
+                pixelHandler(.v1MigrationSuccessful)
 
-            defer {
-                completion(isSubscriptionActive)
+                // cleaning up old data
+                clearSubscriptionCache()
             }
+        } catch {
+            Logger.subscription.error("Failed to migrate V1 token: \(error, privacy: .public)")
+            pixelHandler(.v1MigrationFailed)
+        }
 
-            // Refetch and cache subscription
-            switch await subscriptionEndpointService.getSubscription(accessToken: token, cachePolicy: .reloadIgnoringLocalCacheData) {
-            case .success(let subscription):
-                isSubscriptionActive = subscription.isActive
-            case .failure(let error):
-                if case let .apiError(serviceError) = error, case let .serverError(statusCode, _) = serviceError {
-                    if statusCode == 401 {
-                        // Token is no longer valid
-                        accountManager.signOut()
-                        return
-                    }
+        // Fetching fresh subscription
+        if isUserAuthenticated {
+            do {
+                let subscription = try await getSubscription(cachePolicy: .reloadIgnoringLocalCacheData)
+                Logger.subscription.log("Subscription is \(subscription.isActive ? "active" : "not active", privacy: .public)")
+                if subscription.isActive {
+                    pixelHandler(.subscriptionIsActive)
                 }
+            } catch {
+                Logger.subscription.error("Failed to load initial subscription data: \(error, privacy: .public)")
```

Comment:
Shouldn't this error throw? The VPN calls this at startup: what happens if the user is not authenticated?  Can they still connect?